contracts/near/src/lib.rs

```diff
@@ -0,0 +1,437 @@
+        Self {
+            source: UnorderedSet::new(StorageKeys::Source),
+            prices: UnorderedMap::new(StorageKeys::Prices),
+            wormhole,
+            codehash,
+        }
+    }
+
+    /// Add a new (chain, contract) pair to the list of trusted sources for Price Attestations.
+    #[private]
+    pub fn add_source(&mut self, source: Source) {
+        self.source.insert(&source);
+    }
+
+    /// Remove a (chain, contract) pair from the list of trusted sources for Price Attestations.
+    ///
+    /// NOTE: The caller of this function will be refunded the gas cost of the recovered storage
+    /// but the caller may not be the same as the original caller of `add_source`. This means the
+    /// refund goes to the wrong party. This is OK as the cost is low and the calls are rare and
+    /// the bookkeeping to fix this is not worth it.
+    #[private]
+    pub fn del_source(&mut self, source: Source) {
+        self.source.remove(&source);
+    }
+
+    pub fn get_sources(&self) -> Vec<Source> {
+        self.source.iter().collect()
+    }
+
+    /// Process a Batch of Price Attestations.
+    ///
+    /// This receives price updates, verifies them, and stores the attestations in the contract
+    /// state allowing other contracts to query them.
+    #[private]
+    pub fn attest_prices(&mut self, batch: &BatchPriceAttestation) -> usize {
+        let mut count_updates = 0;
+        for price_attestation in &batch.price_attestations {
+            // Verify the PriceAttestation should be accepted, and if so, store it.
+            if update_price_feed_if_new(&mut self.prices, PriceFeed::from(price_attestation)) {
+                count_updates += 1;
+            }
+        }
+        count_updates
+    }
+
+    /// Read the current Price for a given Price Feed, if the price is past the staleness threshold
+    /// the trading status is set to Unknown before returning.
+    pub fn get_price(&self, price_id: Identifier) -> Option<PriceFeed> {
+        self.prices.get(&price_id).and_then(|mut price| {
+            let block_timestamp = env::block_timestamp();
+            let price_timestamp = price.publish_time.try_into().ok()?;
+
+            // - If Price older than STALENESS_THRESHOLD, set status to Unknown.
+            // - If Price newer than now by more than STALENESS_THRESHOLD, set status to Unknown.
+            // - Any other price around the current time is considered valid.
+            if u64::abs_diff(block_timestamp, price_timestamp) > STALENESS_THRESHOLD {
+                price.status = PriceStatus::Unknown;
+            }
+
+            Some(price)
+        })
+    }
+
+    /// Instruction for processing VAA's relayed via Wormhole.
+    ///
+    /// Note that VAA verification requires calling Wormhole so processing of the VAA itself is
+    /// done in a callback handler, see `process_vaa_callback`.
+    #[payable]
+    #[handle_result]
+    pub fn process_vaa(&mut self, vaa: String) -> Result<(), Error> {
```

Comment:
in general we don't want to mention vaa in our public interfaces. can you use the same name as aptos or near?